MoleculeVAE-reImplementation/dataloader.py

```diff
@@ -1,133 +1,272 @@
+    """This is one hot embedding function
+    - Inputs:
+            -smile_data:list of smile string
+            -normalize_size: the maximum number of characters in a single smile string
+        - Outputs:
+            - one_hot_feature: one hot vector
+            - character_index_lookup_dict:"""
+    character_index_lookup_dict = load_character_index_lookup_dict()
+    one_hot_feature = [
+        make_encoding_for_smiles(vec, character_index_lookup_dict, normalize_size)
+        for vec in smile_data
+    ]
+    return one_hot_feature, character_index_lookup_dict
+
+
+def load_character_index_lookup_dict():
+    """This is a dictionary that maps character to a unique index"""
+    if os.path.exists("Dicdata.json"):
+        with open("Dicdata.json") as json_file:
+            character_index_lookup_dict = json.load(json_file)
     else:
-        od = collections.OrderedDict([(a, list(dicset).index(a)) for a in dicset])
-        with open("Dicdata.json", "w") as outfile:
-            json.dump(od, outfile)
-
-        x = [np.concatenate((torch.nn.functional.one_hot(torch.tensor(list(map(lambda a: od[a], [*vec]))), num_classes=len(od)), torch.zeros(normalizeSize - len(vec), len(od))), axis=0) for vec
-             in SmiData]
-
-    return x,od
-def oneHotdecoder(onehotData,dic):
-    dic_swap = {v: k for k, v in dic.items()}
-    return["".join(map(str, list(map(lambda a: dic_swap[a], ids.argmax(-1))))) for ids in onehotData]
-
-#task='do_prop_pred'or'AE_only'
-def Smiles2dataset(params):
-    if params['FromDeepchem']:
-        datasets=loadDataset(params['Data_file'],params['FromDeepchem'],params['featurizer'])
+        character_index_lookup_dict = collections.OrderedDict(
+            [(a, i) for i, a in enumerate(SMILES_CHARACTER_DIC)]
+        )
+    return character_index_lookup_dict
+
+
+def make_encoding_for_smiles(string_list, character_index_lookup_dict, normalize_size):
+    """This is to convert string to one hot vector
+    - Inputs:
+        -string_list:list of smile string
+        -character_index_lookup_dict: a dictionary that maps character to a unique index
+        -normalize_size: the maximum number of characters in a single smile string
+    - Outputs:
+        - onehot_out: one hot vector
+    """
+
+    index_list = list(map(lambda a: character_index_lookup_dict[a], [*string_list]))
+    num_classes = len(character_index_lookup_dict)
+    one_hot_vector = torch.nn.functional.one_hot(
+        torch.tensor(index_list), num_classes=num_classes
+    )
+    padding_size = (
+        0,
+        0,
+        0,
+        normalize_size - len(string_list),
+    )  # pad last dim by 1 on each side
+    onehot_out = F.pad(one_hot_vector, padding_size, "constant", 0)
+    """output shape=[normalize_size,len(character_index_lookup_dict)]"""
+    return onehot_out
+
+
+def smile_to_dataset(params):
+    """This is to load dataset for training"""
+    if params["FromDeepchem"]:
+        datasets = load_dataset(
```

Comment:
Here it doesn't make much sense to call `load_dataset` which then also just calls a single function. Why not call the function inside `load_dataset` directly here?